src/utils/addons/prepare.js

```diff
@@ -0,0 +1,110 @@
+const chalk = require('chalk')
+
+const ADDON_VALIDATION = {
+  EXISTS: 'EXISTS',
+  NOT_EXISTS: 'NOT_EXISTS',
+}
+
+const validateExists = ({ addon, addonName, siteData, log, exit }) => {
+  if (!addon || !addon.id) {
+    log(`Add-on ${addonName} doesn't exist for ${siteData.name}`)
+    log(`> Run \`netlify addons:create ${addonName}\` to create an instance for this site`)
+    exit(1)
+  }
+}
+
+const validateNotExists = ({ addon, addonName, siteData, log, exit }) => {
+  if (addon && addon.id) {
+    log(`The "${addonName} add-on" already exists for ${siteData.name}`)
+    log()
+    const cmd = chalk.cyan(`\`netlify addons:config ${addonName}\``)
+    log(`- To update this add-on run: ${cmd}`)
+    const deleteCmd = chalk.cyan(`\`netlify addons:delete ${addonName}\``)
+    log(`- To remove this add-on run: ${deleteCmd}`)
+    log()
+    exit(1)
+  }
+}
+
+const getCurrentAddon = ({ addons, addonName }) => {
+  const addon = addons.find(addon => addon.service_path === `/.netlify/${addonName}`)
```

Comment:
no need to fix this in this PR, but just this check is not really great and we could check this with `addon.slug == addonName` way.
however, sadly realized that we have another mistake in our swagger.yml that `showService` is using service instance as a ref, so we need to fix this before doing that way.
https://github.com/netlify/open-api/blob/6d77ca696837986aa9518d1843bd00cf8e4c387d/swagger.yml#L1576

I'll create a PR

Reply:
ah actually this one is getting addons with `getServices`, which properly use `service` as a ref, so we could use `addon.slug == addonName` way.
anyways this PR is already a big change and it has been working with this way so far, so no need to improve in this PR

Reply:
`addon.service_slug` it the way to do it :)

Reply:
ahhh you're right, now I'm looking at where this is called, it's coming from `api.listServiceInstancesForSite`, so it's actually service instance instead of service. cool sounds good to me!